### PR TITLE
Siemens Logo as remote relay.

### PR DESCRIPTION
**Pull request by DerWahreKlinki**
Siemens Logo! as a network relay. Unconfigured Logo! starts with modbus registers 800+.
If configured you have to adjust either the modbus registers in Logo! or in OpenEMS

### CODE REVIEW

**Comment by hydroid7** [comment]
Thank you for the nice work. 
Can you please add device specific instructions / debug instructions to the readme file?
Best

**Comment by DerWahreKlinki** [comment]
Hi,
sorry for the late answer. I´ve added a description to adoc and added a few features:
- now you can use different addresses for reading and writing. This could be necessary if you´ve defined some logic on the Logo
- I´ve made the offset-addresses configurable

File: io.openems.edge.io.siemenslogo/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	io.openems.common,\
+	io.openems.edge.bridge.modbus,\
+	io.openems.edge.common,\
+	io.openems.edge.io.api,\
+	slf4j.api
+
+-testpath: \
+	${testpath},\
+	com.ghgande.j2mod
+Bundle-Description: Siemens LOGO! 8 as network relais. Connected via Modbus. Relays starting \
```
Comment (hydroid7) [inline io.openems.edge.io.siemenslogo/bnd.bnd:17]: Please remove the TODO from the description.

**Review by hydroid7 — CHANGES_REQUESTED** [review]
Thank you for the changes, they LGTM.

Please run eclipse formatter and checkstyle again, as I spotted rule violations here and there.

**Comment by DerWahreKlinki** [comment]
done.
Hope, it´s OK now

**Comment by hydroid7** [comment]
Yes, `Config.java` seems to have a wrong package declaration.

File: io.openems.edge.io.siemenslogo/src/io/openems/edge/io/siemenslogo/Config.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package io.openems.edge.meter.EMH;
+
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import io.openems.edge.meter.api.MeterType;
+
+@ObjectClassDefinition(//
+		name = "io.openems.edge.meter.EMH", //
```
Comment (hydroid7) [inline io.openems.edge.io.siemenslogo/src/io/openems/edge/io/siemenslogo/Config.java:8]: Wrong package name.

File: io.openems.edge.io.siemenslogo/src/io/openems/edge/io/siemenslogo/Config.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+	@AttributeDefinition(name = "Alias", description = "Human-readable name of this Component; defaults to Component-ID")
+	String alias() default "";
+
+	@AttributeDefinition(name = "Is enabled?", description = "Is this Component enabled?")
+	boolean enabled() default true;
+
+	@AttributeDefinition(name = "Meter-Type", description = "What is measured by this Meter?")
+	MeterType type() default MeterType.PRODUCTION;
+
```
Comment (hydroid7) [inline io.openems.edge.io.siemenslogo/src/io/openems/edge/io/siemenslogo/Config.java:23]: Thats not a meter.

**Comment by DerWahreKlinki** [comment]
That was the wrong config.java. I´m sorry for that.

I definetly have to improve my github-workflows.

But the build still fails. I`ll take a deeper look..

**Comment by hydroid7** [comment]
Almost done! :)
The messages are a lot of checkstyle violations.

You can run the checks with gradle:
```bash
$ ./gradlew :io.openems.edge.io.siemenslogo:checkstyleMain
```
That should be faster than pushing to github.

**Comment by DerWahreKlinki** [comment]
Thanks for that advise! Now the build checks are on green light

**Review by hydroid7 — APPROVED** [review]

**Comment by hydroid7** [comment]
Thank you for that well documented module.

**Comment by sfeilmeier** [comment]
Sorry for taking so much time with reviews. I am now trying to distribute the task to different experienced OpenEMS software developers - but most of them are FENECON employees and there are also other tasks to do...

You don't have to close and reopen a PR. It's common practice to just do a "force push" on an existing branch. This also helps to keep the history of a PR.

**Comment by DerWahreKlinki** [comment]
Don´t bother... I haven´t enough time to work on openems constantly either. 
You´ve told me before to "force push" a PR. Such things go out of the brain after a while...

Thanks for keeping up my requests!

**Pull request closed without merging**